File: 2000/LC 1409.cpp
class Solution {
public:
    vector<int> processQueries(vector<int>& queries, int m) {
        int n = queries.size();
        vector<int> pos(m), ans;
        iota(pos.begin(), pos.end(), 1);
        for (int x : queries) {
            auto it = find(pos.begin(), pos.end(), x);
            ans.push_back(it - pos.begin());
            int t = *it;
            pos.erase(it);
            pos.insert(pos.begin(), t);
        }
        return ans;
    }
};

File: 2000/LC 1409.go
func processQueries(queries []int, m int) []int {
    n := len(queries)
    f := make([]int, n + m + 1)
    
    update := func(idx, val int) {
        for i := idx; i < len(f); i += i & -i {
            f[i] += val
        }
    }

    query := func(idx int) (res int) {
        for i := idx; i > 0; i -= i & -i {
            res += f[i]
        }
        return
    }

    pos := make([]int, m + 1)
    for i := 1; i <= m; i++ {
        pos[i] = n + i
        update(n + i, 1)
    }

    ans := make([]int, n)
    for i, x := range queries {
        update(pos[x], -1)
        ans[i] = query(pos[x])
        pos[x] = n - i
        update(pos[x], 1)
    }

    return ans
}